pkg/logproto/compat_test.go

```diff
@@ -166,3 +166,206 @@ func TestLegacyLabelPairCompatibilityUnmarshalling(t *testing.T) {
 	require.NoError(t, err)
 	require.NotEqualValues(t, expectedLabelPair, incompatibleLabelPair)
 }
+
+func TestMergeLabelResponses(t *testing.T) {
```

Comment:
Nice work on the tests. Could you add cases for `Label` and `Series` as well? The Mimir test give some test cases to verify that `__tenant_id__` is included (https://github.com/grafana/mimir/blob/main/pkg/querier/tenantfederation/merge_queryable_test.go#L560-L869).

Reply:
Yep! I've included them in `multi_tenant_querier_test.go`